src/pages/CDP.js

```diff
@@ -210,6 +210,7 @@ function CDPView({ cdpId: _cdpId }) {
   }, [account]);
 
   useEffect(() => {
+    let didCancel = false;
     (async () => {
       const cdpManager = maker.service('mcd:cdpManager');
       const cdp = await cdpManager.getCdp(cdpId);
```

Comment:
@levity Since `getCdp` is async, do we need an `if (didCancel) return` check after it resolves to prevent the possibly unnecessary async call that follows?

Reply:
We could, but it’s not strictly necessary. The main thing is to skip trying to update state with setCdp. It shouldn’t make any more remote calls anyway because data is cached in the cdp object.

Reply:
> It shouldn’t make any more remote calls anyway because data is cached in the cdp object.

Will this be the case if 3 successive requests to `cdpManager.getCdp(cdpId)` are immediately fired off (for the first time for a particular CDP)?

If you look through the requests in the network tab, you'll see there are quite a few duplicates being made (to the same contracts with the same calldata). An easy way to see which are duplicates is to select the response tab and then down arrow through the requests.